File: Proyecto_Tienda/src/main/java/com/jvillacorta/Proyecto_Tienda/models/entity/Proveedor.java
package com.jvillacorta.Proyecto_Tienda.models.entity;

import java.io.Serializable;
import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.PrePersist;
import javax.persistence.PreUpdate;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import org.springframework.format.annotation.DateTimeFormat;

@Entity
@Table(name = "proveedor")
public class Proveedor implements Serializable{

	private static final long serialVersionUID = 1L;
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name="idproveedor")
	private Long id;
	
	@Column(name="ruc")
	private String ruc;
	
	@Column(name="razonsocial")
	private String razonSocial;
	
	@Column(name="telefono")
	private String telefono;
	
	@Column(name="correo")
	private String correo;
	
	@Column(name = "direccion")
	private String direccion;
	
	@Column(name="activo")
	private String estado;
	
	@Column(name="fecharegistro")
	@Temporal(TemporalType.DATE)
	@DateTimeFormat(pattern = "yyyy-MM-dd")
	private Date fechaCreacion;
	
	@PrePersist
	@PreUpdate
	public void prePersist() {
		fechaCreacion = new Date();
	}

	//GETTER Y SETTER
	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getRuc() {
		return ruc;
	}

	public void setRuc(String ruc) {
		this.ruc = ruc;
	}

	public String getRazonSocial() {
		return razonSocial;
	}

	public void setRazonSocial(String razonSocial) {
		this.razonSocial = razonSocial;
	}

	public String getTelefono() {
		return telefono;
	}

	public void setTelefono(String telefono) {
		this.telefono = telefono;
	}

	public String getCorreo() {
		return correo;
	}

	public void setCorreo(String correo) {
		this.correo = correo;
	}

	public String getDireccion() {
		return direccion;
	}

	public void setDireccion(String direccion) {
		this.direccion = direccion;
	}

	public String getEstado() {
		return estado;
	}

	public void setEstado(String estado) {
		this.estado = estado;
	}

	public Date getFechaCreacion() {
		return fechaCreacion;
	}

	public void setFechaCreacion(Date fechaCreacion) {
		this.fechaCreacion = fechaCreacion;
	}
}

File: Proyecto_Tienda/src/main/resources/static/js/proveedor.js
$(document).ready(function(){
	$('body').on('click','.editarProveedor', function(){
		$('#id').val($(this).attr('id'));
		$('#ruc').val($(this).attr('ruc'));
		$('#razonSocial').val($(this).attr('razon_social'));
		$('#telefono').val($(this).attr('telefono'));
		$('#correo').val($(this).attr('correo'));
		$('#direccion').val($(this).attr('direccion'));
		
		const variableEstado = $(this).attr('estado');
		let estado;
		if(variableEstado == "Activo"){
			estado = 1;
		} else {
			estado = 0;
		}
		$('#estado').val(estado);
	})
});

$(document).ready(function(){
	$('body').on('click','.nuevoProveedor', function(){
		$('#id').val($(this).attr(''));
		$('#ruc').val($(this).attr(''));
		$('#razonSocial').val($(this).attr(''));
		$('#telefono').val($(this).attr(''));
		$('#correo').val($(this).attr(''));
		$('#direccion').val($(this).attr(''));
		$('#estado').val($(this).attr(''));
	})
});